agent/engine/execcmd/manager_start_windows_test.go

```diff
@@ -0,0 +1,477 @@
+						{
+							Name: ExecuteCommandAgentName,
+							ManagedAgentState: apicontainer.ManagedAgentState{
+								ID: testUUID,
+							},
+						},
+					}
+				}
+			}
+
+			times := maxRetries
+			retryableErr, isRetryable := test.expectedError.(errors2.RetriableError)
+			if test.expectedError == nil || (isRetryable && !retryableErr.Retry()) {
+				times = 1
+			}
+			if test.expectCreateContainerExec {
+				execCfg := types.ExecConfig{
+					User:   "NT AUTHORITY\\SYSTEM",
+					Detach: true,
+					Cmd:    []string{"C:\\Program Files\\Amazon\\SSM\\amazon-ssm-agent.exe"},
+				}
+				client.EXPECT().CreateContainerExec(gomock.Any(), testTask.Containers[0].RuntimeID, execCfg, dockerclient.ContainerExecCreateTimeout).
+					Return(test.createContainerExecRes, test.createContainerExecErr).
+					Times(times)
+			}
+
+			if test.expectStartContainerExec {
+				client.EXPECT().StartContainerExec(gomock.Any(), testDockerExecId, dockerclient.ContainerExecStartTimeout).
+					Return(test.startContainerExecErr).
+					Times(times)
+			}
+
+			if test.expectInspectContainerExec {
+				client.EXPECT().InspectContainerExec(gomock.Any(), testDockerExecId, dockerclient.ContainerExecInspectTimeout).
+					Return(test.inspectContainerExecRes, test.inspectContainerExecErr).
+					Times(times)
+			}
+
+			mgr := newTestManager()
+			prevMetadata := getAgentMetadata(test.containers[0])
+			err := mgr.StartAgent(context.TODO(), client, testTask, testTask.Containers[0], testTask.Containers[0].RuntimeID)
+			if test.expectedError != nil {
+				assert.Equal(t, test.expectedError, err, "Wrong error returned")
+				// When there's an error, ExecCommandAgentMetadata should not be modified
+				newMetadata := getAgentMetadata(test.containers[0])
+				assert.Equal(t, prevMetadata, newMetadata)
+			} else { // No error case
+				assert.NoError(t, err, "No error was expected")
+				if !test.execEnabled {
+					_, ok := test.containers[0].GetManagedAgentByName(ExecuteCommandAgentName)
+					assert.False(t, ok)
+				} else {
+					ma, _ := test.containers[0].GetManagedAgentByName(ExecuteCommandAgentName)
+					execMD := getAgentMetadata(test.containers[0])
+					assert.Equal(t, strconv.Itoa(testPid1), execMD.PID, "PID not equal")
+					assert.Equal(t, testDockerExecId, execMD.DockerExecID, "DockerExecId not equal")
+					assert.Equal(t, "C:\\Program Files\\Amazon\\SSM\\amazon-ssm-agent.exe", execMD.CMD)
+					assert.Equal(t, test.expectedStatus, ma.Status, "Exec status not equal")
+					assert.WithinDuration(t, test.expectedStartTime, ma.LastStartedAt, 5*time.Second, "StartedAt not equal")
+				}
+			}
+		})
+	}
+}
+
+func TestIdempotentStartAgent(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	client := mock_dockerapi.NewMockDockerClient(ctrl)
+	const (
+		testDockerExecId = "abc"
+		testPid          = 111
+	)
+	var (
+		testPidStr = strconv.Itoa(testPid)
+		testCmd    = "C:\\Program Files\\Amazon\\SSM\\amazon-ssm-agent.exe"
```

Comment:
@angelcar this is 1 example of where the hard coded string would differ Linux/Windows. Theres like 4-5 places with this, other than that the test cases are the exact same

Reply:
could you make these constants in the platform specific test files instead of duplicating the test code?